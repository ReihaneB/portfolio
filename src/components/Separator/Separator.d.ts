export interface SeparatorProps {
  text?: string | null;
}
