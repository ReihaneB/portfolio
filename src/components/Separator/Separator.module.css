.root {
  border: none;
  border-top: 1px solid var(--grey);
  color: var(--grey);
  overflow: visible;
  text-align: center;
}

.root::before {
  background: var(--background);
  content: var(--separator-text);
  padding-left: 16px;
  padding-right: 16px;
  position: relative;
  top: -10px;
}
