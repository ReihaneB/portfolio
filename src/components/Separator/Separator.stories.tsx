import type { Meta, StoryObj } from '@storybook/react';
import { expect, userEvent, waitFor, within } from '@storybook/test';

import { lokiPlayDecorator } from '@/testUtils/loki/play-decorator';

import Separator from './Separator';

const meta: Meta<typeof Separator> = {
  title: 'Components/Separator',
  component: Separator,
};

export default meta;
type Story = StoryObj<typeof Separator>;

const defaultArgs = {
  text: 'Seperator content',
};

export const Playground: Story = {
  args: {
    ...defaultArgs,
  },
};

export const WithoutChildren: Story = {
  args: {
    ...defaultArgs,
    text: null,
  },
};

export const NotFocusTab: Story = {
  args: {
    ...defaultArgs,
  },
  play: lokiPlayDecorator(
    async ({ canvasElement }) => {
      const canvas = within(canvasElement);

      await userEvent.tab();

      await waitFor(() => expect(canvas.getByTestId('separator-component')).not.toHaveFocus());
    }
  ),
};
