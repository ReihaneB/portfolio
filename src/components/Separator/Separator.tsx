import React, { memo } from 'react';

import styles from './Separator.module.css';
import type { SeparatorProps } from './Separator.d';

function Separator({
  text = '',
  ...rest
}: SeparatorProps) {
  const style = text ? (
    { '--separator-text': `"${text}"` } as React.CSSProperties
  ) : undefined;

  return (
    <hr
      className={styles.root}
      data-testid="separator-component"
      style={style}
      {...rest}
    />
  );
}

export default memo(Separator);
