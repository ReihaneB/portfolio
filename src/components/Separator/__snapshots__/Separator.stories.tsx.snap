// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Separator NotFocusTab play-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <hr class="src-components-Separator-Separator-module__root--N5eOl"
      data-testid="separator-component"
      style="--separator-text: &quot;Seperator content&quot;;"
  >
</main>
`;

exports[`Components/Separator Playground smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <hr class="src-components-Separator-Separator-module__root--N5eOl"
      data-testid="separator-component"
      style="--separator-text: &quot;Seperator content&quot;;"
  >
</main>
`;

exports[`Components/Separator WithoutChildren smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <hr class="src-components-Separator-Separator-module__root--N5eOl"
      data-testid="separator-component"
  >
</main>
`;
